pkg/expression/builtin_other.go

```diff
@@ -1676,3 +1678,60 @@ func (b *builtinGetParamStringSig) evalString(ctx EvalContext, row chunk.Row) (s
 	}
 	return str, false, nil
 }
+
+// getQueryAttrFunctionClass for plan cache of prepared statements
+type getQueryAttrFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *getQueryAttrFunctionClass) getFunction(ctx BuildContext, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETString, types.ETString)
+	if err != nil {
+		return nil, err
+	}
+	bf.tp.SetFlen(mysql.MaxFieldVarCharLength)
+	sig := &builtinGetQueryAttrStringSig{baseBuiltinFunc: bf}
+	return sig, nil
+}
+
+type builtinGetQueryAttrStringSig struct {
+	baseBuiltinFunc
+	expropt.SessionVarsPropReader
+}
+
+func (b *builtinGetQueryAttrStringSig) Clone() builtinFunc {
+	newSig := &builtinGetQueryAttrStringSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+func (b *builtinGetQueryAttrStringSig) RequiredOptionalEvalProps() OptionalEvalPropKeySet {
+	return b.SessionVarsPropReader.RequiredOptionalEvalProps()
+}
+
+func (b *builtinGetQueryAttrStringSig) evalString(ctx EvalContext, row chunk.Row) (string, bool, error) {
```

Comment:
```suggestion
// This implements `mysql_query_attribute_string(str)`
func (b *builtinGetQueryAttrStringSig) evalString(ctx EvalContext, row chunk.Row) (string, bool, error) {
```

Reply:
Often people know the function name and then have to fine the location of the code. This might help to make it easier to find.